prov/sm2/src/sm2_atomic.c

```diff
@@ -81,8 +81,7 @@ sm2_atomic_format(struct sm2_xfer_entry *xfer_entry, uint8_t datatype,
 		xfer_entry->hdr.size = ofi_copy_from_iov(
 			atomic_entry->atomic_data.buf,
 			SM2_ATOMIC_COMP_INJECT_SIZE, iov, iov_count, 0);
-		comp_size = ofi_copy_from_iov(atomic_entry->atomic_data.comp +
```

Comment:
is hdr part of the data to be copied?

Reply:
No, this is just copying the compare data into the atomic entry buffer. The hdr shouldn't be getting touched here.

Reply:
So `atomic_entry->atomic_data.comp` doesn't have any hdr actually?

Reply:
Right, the atomic entry has a hdr but the data buffers do not:
```
struct sm2_atomic_hdr {
	uint8_t datatype;
	uint8_t atomic_op;
	int rma_ioc_count;
	struct fi_rma_ioc rma_ioc[SM2_IOV_LIMIT];
	int result_iov_count;
	struct iovec result_iov[SM2_IOV_LIMIT];
};

struct sm2_atomic_data {
	union {
		uint8_t data[SM2_ATOMIC_INJECT_SIZE];
		struct {
			uint8_t buf[SM2_ATOMIC_COMP_INJECT_SIZE];
			uint8_t comp[SM2_ATOMIC_COMP_INJECT_SIZE];
		};
	};
};

struct sm2_atomic_entry {
	struct sm2_atomic_hdr atomic_hdr;
	struct sm2_atomic_data atomic_data;
};
```